app/models/name.rb

```diff
@@ -408,33 +408,95 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  include CreatedUpdatedScopes
 
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
```

Comment:
Thank you for meticulously keeping the extra space in one case but not the other!  I wanted lichen to match lichenicolous fungus, as well, but lichen-haters may well want to see lichenicolous fungi, especially ones that are not obligate to lichens, so it should only reject the exact match "lichen" not matches which contain "lichen".